src/power_grid_model/validation/rules.py

```diff
@@ -697,6 +698,38 @@ def none_missing(data: SingleDataset, component: str, fields: Union[str, List[st
     return errors
 
 
+def valid_p_q_sigma(data: SingleDataset, component: str) -> List[MultiFieldValidationError]:
+    """
+    Check validity of the pair `(p_sigma, q_sigma)` for 'sym_power_sensor' and 'asym_power_sensor'.
+
+    Args:
+        data: The input/update data set for all components
+        component: The component of interest, in this case only 'sym_power_sensor' or 'asym_power_sensor'
+
+    Returns:
+        A list containing zero or one MultiFieldValidationError, listing the p_sigma and q_sigma mismatch.
+        Note that with asymetric power sensors, partial assignment of p_sigma and q_sigma is also considered mismatch.
+    """
+    errors = []
+    p_sigma = data[component]["p_sigma"]
+    q_sigma = data[component]["q_sigma"]
+    p_nan = np.isnan(p_sigma)
+    q_nan = np.isnan(q_sigma)
+    p_inf = np.isinf(p_sigma)
+    q_inf = np.isinf(q_sigma)
+    if p_sigma.ndim > 1:  # if component == 'asym_power_sensor':
+        p_nan = p_nan.any(axis=-1)
+        q_nan = q_nan.any(axis=-1)
+        p_inf = p_inf.any(axis=-1)
+        q_inf = q_inf.any(axis=-1)
+    mis_match = p_nan != q_nan
+    mis_match |= p_inf.any() or q_inf.any()
+    if mis_match.any():
+        ids = [data[component]["id"].flatten().tolist()]
```

Comment:
```suggestion
        ids = [data[component]["id"][mis_match].flatten().tolist()]
```

please double-check with my proposed change from my earlier comment